domain/secret/service/delete.go

```diff
@@ -86,8 +87,11 @@ func (s *SecretService) removeFromBackend(
 			return errors.Annotatef(err, "deleting secret content from backend for %q", revId)
 		}
 	}
-	if err := p.CleanupSecrets(ctx, &cfg, uri, revs); err != nil {
-		return errors.Annotate(err, "cleaning secret resources from backend")
+	// For units, we want to clean up any backend artefacts.
+	if accessor.Kind == UnitAccessor {
+		if err := p.CleanupSecrets(ctx, &cfg, accessor.ID, revs); err != nil {
+			return errors.Annotate(err, "cleaning secret resources from backend")
```

Comment:
We can include the accessor.ID and p.Type() in the error message.

Reply:
good idea